clinica/pydra/interfaces.py

```diff
@@ -32,35 +36,59 @@ def __init__(self, **kwargs):
         # used for mandatory inputs check
         undefined_traits = {}
         self.inputs.trait_set(trait_change_notify=False, **undefined_traits)
+        self.sessions = None
+        self.subjects = None
 
     def _list_outputs(self):
-        from clinica.utils.inputs import clinica_file_reader, clinica_group_reader
         from clinica.utils.participant import get_subject_session_list
 
         sessions, subjects = get_subject_session_list(
             self.inputs.base_dir,
             is_bids_dir=False,
         )
-        query = {}
-        for k, q in self.inputs.output_query.items():
-            reader = q.pop("reader")
-            if reader == "file":
-                query[k] = clinica_file_reader(
-                    subjects,
-                    sessions,
-                    self.inputs.base_dir,
-                    q,
-                )
-            elif reader == "group":
-                query[k] = clinica_group_reader(self.inputs.base_dir, q)
+        self.sessions = sessions
+        self.subjects = subjects
+
+        output_query = {}
+        for k, query in self.inputs.output_query.items():
+            if isinstance(query, list):
+                temp = [self._execute_single_query(q) for q in query]
+                if len(temp) != len(self.subjects) and len(temp[0]) == len(
+                    self.subjects
+                ):
+                    temp = [list(i) for i in zip(*temp)]  # Transpose
+                output_query[k] = temp
             else:
-                raise ValueError(f"Unknown reader {reader}.")
-        return query
+                output_query[k] = self._execute_single_query(query)
+        return output_query
+
+    @abc.abstractmethod
+    def _execute_single_query(self, query: Dict) -> List[str]:
```

Comment:
Just a comment, if you are not going to type the dictionary (or list) further, just use `dict` (or `list`) instead.